system/lib/emmalloc.cpp

```diff
@@ -354,10 +354,23 @@ static void resize_region(void *ptr, uint32_t size)
   assert(HAS_ALIGNMENT(ptr, sizeof(uint32_t)));
   assert(HAS_ALIGNMENT(size, sizeof(uint32_t)));
   assert(size >= sizeof(Region));
+
+  // We are resizing an existing region that is either in free or used state. The newly resized region
+  // should preserve this free/used status, so look at the existing region and its in-use status,
+  // and propagate the state to the new region.
+  uint32_t oldSize = (int32_t)region_ceiling_size((Region*)ptr);
+
+  // Free regions have all the size bits reversed (highest bit tells it is a free region)
+  uint32_t usedMask = (uint32_t)(((int32_t)oldSize) >> 31); // Sign extend the highest bit to whole word
+
+  // Write new region size:
   *(uint32_t*)ptr = size;
-  uint32_t *sizeAtEnd = (uint32_t*)ptr + (size>>2) - 1;
-  uint32_t usedMask = (*(int32_t*)sizeAtEnd) >> 31;
-  *sizeAtEnd = size ^ usedMask;
+  uint32_t *newRegionEnd = (uint32_t*)((uintptr_t)ptr + size);
+  newRegionEnd[-1] = size ^ (usedMask >> 31);
```

Comment:
`usedMask` is either all 0s or all 1s, I believe? If so then `usedMask >> 31` is either 0 or 1. Is that right? That seems to affect only the lowest bit in the XOR here.

Reply:
Err, you are absolutely right.. had an excess >> 31 there.

However now that I look at this with fresh eyes in the morning and wonder why this ever worked, I realize that this function was originally used to resize both free and used regions, but after a change at some point, this function is now only ever called to resize used regions. That is why everything passes with this PR: `usedMask` is always all zeros.

Hence we can simplify and just nuke the whole function..